streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -219,11 +219,7 @@ private boolean waitOnState(final State targetState, final long waitMs) {
             long elapsedMs = 0L;
             while (state != targetState) {
                 if (waitMs == 0) {
-                    try {
-                        stateLock.wait();
-                    } catch (final InterruptedException e) {
-                        // it is ok: just move on to the next iteration
-                    }
+                    return false;
```

Comment:
This seems to be the same condition as the `else` block, and actually, given that `waitMs` can't be negative, I think the `else` block is unreachable. Perhaps we can consolidate the logic?